dvc/repo/experiments/init.py

```diff
@@ -258,13 +258,19 @@ def init(
         cmd=context["cmd"],
         deps=compact([context.get("code"), context.get("data")]),
         params=[{params: None}] if params else None,
-        metrics_no_cache=compact([context.get("metrics"), live_metrics]),
-        plots_no_cache=compact([context.get("plots"), live_plots]),
         force=force,
         **{
             "checkpoints"
```

Comment:
I think the common way:

```python
if type == "checkpoint":
    outs = "checkpoint"
    ...
else:
    outs = "outs"

{
outs: compact([models]),
...
}
```

Are more readable

Reply:
And I'm sorry that why couldn't we reuse the field name, and only add a checkpoint flag in it?

Reply:
> And I'm sorry that why couldn't we reuse the field name, and only add a checkpoint flag in it?

We can't because it is not how the internal `repo.stage.create` expects the arguments (https://github.com/iterative/dvc/blob/0b073543611db4abd24c75d3b55abad6850b2111/dvc/stage/utils.py#L45)

Reply:
> Are more readable

Updated